tools/CMakeLists.txt

```diff
@@ -34,6 +35,7 @@ kwiver_add_library( kwiver_tools_applet
 target_link_libraries( kwiver_tools_applet
     PUBLIC    vital_util
               ${VITAL_BOOST_REGEX}
+              ${Boost_REGEX_LIBRARY}
```

Comment:
Is this really needed?  We've been trying to get rid of Boost.  I though @linus-sherrill had replaced Boost regex with something else?

Reply:
I believe so. It is originating from https://github.com/Kitware/kwiver/blob/master/tools/cxxopts.hpp#L32 which causes errors when only vital and tools are enabled.

Reply:
This was an unsatisfactory solution to a difficult problem. The std::regex is broken on the default compiler for CentOS7, so I substituted the boost regex package only when std::regex can not handle the required patterns. The longer term solution will be to get rid of this hack when we can move past the g++ with the deficient compiler library.

Reply:
Is there a way to make this dependency on boost conditional on whether you have the bad compiler?  Actually, tt might be better if `${VITAL_BOOST_REGEX}` depended brought in this dependency if needed.

What I'm trying to avoid is KWIVER always depending on Boost, even when it is not really needed.  If I don't have the bad compiler then I should be able to build this without Boost.